app/lib/widgets/wallets/contract_details.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:flutter/material.dart';
+import 'package:gridproxy_client/models/contracts.dart';
+import 'package:threebotlogin/helpers/logger.dart';
+import 'package:threebotlogin/helpers/contract_helpers.dart';
+
+class ContractDetails extends StatelessWidget {
+  final ContractInfo contract;
+
+  const ContractDetails({
+    super.key,
+    required this.contract,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    try {
+      List<Widget> detailRows = [];
+      final contractDetails = extractContractDetails(contract);
```

Comment:
what if the function returned an empty map ?

Reply:
ExtractContractDetails can't return an empty map as it always populates the first 5 fields.

https://github.com/threefoldtech/threefold_connect/blob/464959723d7135b0f17debbf06b5ad45ba2a3ee7/app/lib/helpers/contract_helpers.dart#L54-L58